### DISCUSSION TITLE
Display progress message to user when running py-editor the first time.

### DISCUSSION THREAD

**Discussion by MarcSkovMadsen** [pyscript/pyscript#2121] [Proposals]
I would like to be able to use the `py-editor` with Panel. Currently it takes a long time to run the editor the first time run is clicked.

To minimize the feeling of waiting time and/ or confusion I would like to request progress messages are shown - not just a spinner. Every other use of pyodide or pyscript I've seen does this. Please also do this for the editor.

As you can see below there is a long waiting time. And no only a loading spinner is displayed.

https://github.com/user-attachments/assets/f42f6ea8-c79c-4bce-900c-8af4ba9b3b1c

**Comment by WebReflection** [comment]
The time is unpredictable because it depends on packages, their dependencies, network latency, network speed, and so on.

We can't provide any valuable detail/info there and the logs you see come from the worker, not the main thread, so we can't temporarily override those logs neither.

What you could do when you know it's going to take long time is to play with the button CSS clicked / spinning state and enhance it or animate it in a way that is more clear but any other detail behind the scene might be misleading.

In other words: what do you expect to see?

**Reply by MarcSkovMadsen** [reply]
I would expect to see an improved version of what is shown in the browser console and then a little bit more info like loading of external packages and finally starting/ running after everything has loaded.

Panel does this very well for for apps converted to Pyodide and for the Pyodide sphinx code embedding. Also when using web workers.

**Reply by WebReflection** [reply]
we might consider forwarding somehow printed messages while awaiting packages to be installed but I am not sure we should be obtrusive on the UI side ... so far you are the first asking for such details so there's work to be done there, after discussing this with the rest of the team too.

**Reply by WebReflection** [reply]
To be super clear: we don't get to dictate what Pyodide prints or not while bootstrapping with packages and the rest, so having anything not provided out of the box seems to be unnecessary ... I hope that's fine.

**Comment by MarcSkovMadsen** [comment]
What I'm looking for are progress messages like Panel displays https://panel.holoviz.org/getting_started/build_app.html. They are super helpful for users such that they know things are happening and somethings coming.

https://github.com/user-attachments/assets/b8ee49e9-0dce-46a3-955d-918b1085ce76

**Reply by WebReflection** [reply]
nice but ... so many questions:

  * why are other editors also running automatically?
  * why aren't other editors showing anything at all?
  * what if you click just the second editor, not the first?

Anyway, those details are mostly out of what Pyodide logs in console while bootstrapping ... once again, we don't catch those details so we need to think how to do so and expose these over the editor somehow. I think it's doable but the rest of the flow in that video is a bit obscure to me.

**Comment by MarcSkovMadsen** [comment]
Another example is Panel converted apps. For example like https://awesome-panel-video-stream.static.hf.space/index.html.
